### PR TITLE
Test that trace and span id are present where they are expected by the product

### PR DESCRIPTION
**Pull request by vladkluev**
# Context
We currently store trace_id and span_id in two places on the errors table:

* As columns
* As keys and values on the contexts table

The product expects both of these to be present

# Solution
Add tests to make sure that if someone removes these it's intentional

### CODE REVIEW

File: snuba/datasets/errors_processor.py [hunk 1]
```diff
@@ -113,3 +108,8 @@ def extract_promoted_contexts(
             output["trace_id"] = str(uuid.UUID(transaction_ctx["trace_id"]))
         if transaction_ctx.get("span_id", None):
             output["span_id"] = int(transaction_ctx["span_id"], 16)
+        # We store trace_id and span_id as promoted columns and on the query level
+        # we make sure that all queries on contexts[trace.trace_id/span_id] use those promoted
+        # columns instead. So we don't need to store them
+        transaction_ctx.pop("trace_id", None)
+        transaction_ctx.pop("span_id", None)
```
Comment (lynnagara) [inline snuba/datasets/errors_processor.py:115]: I'd write this as
```
trace_id = transaction_ctx.pop("span_id", None)
if trace_id is not None:
   output["trace_id"] = str(uuid.UUID(trace_id))
```

**Review by lynnagara — COMMENTED** [review]
This seems good to me. Just one question - do you know whether we are returning trace ID data from contexts anywhere? I noticed we have this in Sentry and I'm wondering if this change is going to alter some of the results we're sending back https://github.com/getsentry/sentry/blob/f718040568a10f566db096e4b6202be7d133fd44/src/sentry/snuba/events.py#L260-L265

**Comment by vladkluev** [comment]
That URL is for the `events` dataset which is not altered.

**Comment by vladkluev** [comment]
But also I don't really have good context for the sentry codebase. As far as I know, clickhouse is only used for querying. We don't return clickhouse rows, we just return event ids to look up in nodestore right?

**Review by lynnagara — APPROVED** [review]
The events dataset is the main way we query the Clickhouse errors table, and we can return many columns not just event ID. If we change the data we're writing to some columns then queries to that dataset will return slightly different results. I don't think that particular column is important though so this is not a blocker to me. I think there is a chance this may slightly change some results we send to Sentry in some edge cases though.

**Review by fpacifici — CHANGES_REQUESTED** [review]
Please do not remove these columns from the errors table without having ensured nobody is using them. We validated that nobody is using them via the querylog only for transactions. Unless somebody else did the test for you we would have to test the queries against the errors table as well. 

Specifically, issue search allows customers to issue arbitrary queries on the contexts column (I suspect this is basically unintended tech debt, but it is now in use). 

The task was aimed at reducing the size of contexts on the transactions table. The errors table is not a problem at present. So I would focus first on transactions, measure the improvement and then we could get to the errors table if needed after testing that we are never querying those contexts via arrayJoin as well.

**Comment by vladkluev** [comment]
@fpacifici since it actually is important I'll remove the storage changes but I'll keep the tests that make sure they're there.

File: snuba/datasets/errors_processor.py [hunk 2]
```diff
@@ -109,7 +104,10 @@ def extract_promoted_contexts(
...
-            output["trace_id"] = str(uuid.UUID(transaction_ctx["trace_id"]))
-        if transaction_ctx.get("span_id", None):
-            output["span_id"] = int(transaction_ctx["span_id"], 16)
+        trace_id = transaction_ctx.get("trace_id", None)
+        span_id = transaction_ctx.get("span_id", None)
+
+        if trace_id:
+            output["trace_id"] = str(uuid.UUID(trace_id))
+        if span_id:
+            output["span_id"] = int(span_id, 16)
```
Comment (fpacifici) [inline snuba/datasets/errors_processor.py:113]: You do not need this change anymore. Up to you.

**Review by fpacifici — APPROVED** [review]
thanks

File: snuba/datasets/errors_processor.py (hunk 2 above)
Comment (vladkluev) [reply snuba/datasets/errors_processor.py:113]: technically it's a performance improvement 😂

**Pull request merged**